### PR TITLE
Bumped OpenTelemetry Collector to v0.19.0

### LINKED ISSUE #176: Update to collector v0.19.0
Same as #160, but for 0.19.0. Changelog: https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.19.0

### PR DESCRIPTION
**Pull request by jpkrohling**
Closes #176

﻿Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Changes by Version
 ==================
 
+0.19.0 (2021-01-27)
+-------------------
+* Bumped OpenTelemetry Collector to v0.19.0 ([#nn](https://github.com/open-telemetry/opentelemetry-operator/pull/nn), [@jpkrohling](https://github.com/jpkrohling))
```
Comment (objectiser) [inline CHANGELOG.md:6]: nit: Does it need to refer to itself? Maybe in these circumstances it should just have a line that says "Bumping version of OpenTelemetry Collector to v0.19.0 - no other changes"

File: pkg/collector/upgrade/v0_19_0.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		// Remove deprecated configs from resource processor: type (set "opencensus.type" key in "attributes.upsert" map instead) and labels (use "attributes.upsert" instead).
+		if strings.HasPrefix(k.(string), "resource") {
+			switch processor := v.(type) {
+			case map[interface{}]interface{}:
+				// type becomes an attribute.upsert with key opencensus.type
+				if typ, found := processor["type"]; found {
+					var attributes []map[string]string
+					if attrs, found := processor["attributes"]; found {
+						if attributes, ok = attrs.([]map[string]string); !ok {
+							otelcol.Status.Messages = append(otelcol.Status.Messages, fmt.Sprintf("upgrade to v0.19.0 reset the attributes list for processors %q as it couldn't parse the previous value. Type: %t, value: %v", k, attrs, attrs))
```
Comment (objectiser) [inline pkg/collector/upgrade/v0_19_0.go:63]: Are you sure this shouldn't fail the upgrade? As the missing attributes are likely going to have an impact on the behaviour of the collector once upgraded - although having said that, if it isn't a `map[string]string`, then probably there are more issues.

File: pkg/collector/upgrade/v0_19_0.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+					delete(processor, "type")
+					otelcol.Status.Messages = append(otelcol.Status.Messages, fmt.Sprintf("upgrade to v0.19.0 migrated the property 'type' for processor %q", k))
+				}
+
+				// handle labels
+				if labels, found := processor["labels"]; found {
+					var attributes []map[string]string
+					if attrs, found := processor["attributes"]; found {
+						if attributes, ok = attrs.([]map[string]string); !ok {
+							otelcol.Status.Messages = append(otelcol.Status.Messages, fmt.Sprintf("upgrade to v0.19.0 reset the attributes list for processors %q as it couldn't parse the previous value. Type: %t, value: %v", k, attrs, attrs))
```
Comment (objectiser) [inline pkg/collector/upgrade/v0_19_0.go:82]: Same comment as above.

**Review by objectiser — COMMENTED** [review]
Just a couple of comments. Also no tests for the error paths?

File: pkg/collector/upgrade/v0_19_0.go (hunk 2 above)
Comment (jpkrohling) [reply pkg/collector/upgrade/v0_19_0.go:63]: Yeah, the idea is that if it's not map, the upgrade might fix an existing issue. But perhaps the upgrade shouldn't try to be too smart. I'll make it fail the upgrade.

**Comment by jpkrohling** [comment]
PR updated

**Review by objectiser — DISMISSED** [review]

**Comment by jpkrohling** [comment]
@objectiser would you mind reviewing it again?

**Review by objectiser — APPROVED** [review]

**Pull request merged**